omegaconf/base.py

```diff
@@ -531,6 +531,9 @@ def _re_parent(self) -> None:
                         if isinstance(item, Container):
                             item._re_parent()
 
+    def _has_ref_type(self) -> bool:
+        return self._metadata.ref_type not in (None, Any)
```

Comment:
When changing the value of a key with primitive container if the said key doesn't have a ref_type assigned (None) it should be set instead of wrapping a new node.
In Any's case, if a dictconfig has an Any ref_type when setting its value with a different type of dict, dictconfig or structured_config it fails so, we need to wrap a new node instead.

Reply:
Please explain with examples. I am not following.

Reply:
Let's say that a container has a ref_type if only it isn't None
```
# foo with none ref_type
cfg = OmegaConf.create({"foo": DictConfig(content={})})
cfg.foo = [1,2] # foo doesn't have a ref_type so it wraps
# foo with any ref_type
ref_type = Any 
cfg = OmegaConf.create({"foo": DictConfig(ref_type=ref_type, content={})})
cfg.foo = [1,2] # here it should wrap but instead, it sets value resulting in an error
```
If I'm understanding this correctly a key that has an Any ref_type, should be able to wrap new nodes independently  of their type. So that's why I'm including Any there, so the second case won't give an error.